modular_skyrat/modules/bluespace_miner/code/bluespace_miner.dm

```diff
@@ -0,0 +1,167 @@
+/obj/machinery/bluespace_miner
+	name = "bluespace miner"
+	desc = "Through the power of bluespace, it is capable of producing materials."
+	icon = 'modular_skyrat/modules/bluespace_miner/icons/bluespace_miner.dmi'
+	icon_state = "miner"
+
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/bluespace_miner
+	idle_power_usage = 200
+
+	///the temperature of the co2 produced per successful process (its really 100)
+	var/gas_temp = 100
+	///the amount of seconds process_speed goes on cooldown for
+	var/processing_speed = 6 SECONDS
+	///the chance each ore has to be picked, weighted list
+	var/list/ore_chance = list(
+		/obj/item/stack/sheet/iron = 20,
+		/obj/item/stack/sheet/glass = 20,
+		/obj/item/stack/sheet/mineral/plasma = 14,
+		/obj/item/stack/sheet/mineral/silver = 8,
+		/obj/item/stack/sheet/mineral/titanium = 8,
+		/obj/item/stack/sheet/mineral/uranium = 3,
+		/obj/item/stack/sheet/mineral/gold = 3,
+		/obj/item/stack/sheet/mineral/diamond = 1,
+	)
+	COOLDOWN_DECLARE(process_speed)
+
+/obj/machinery/bluespace_miner/RefreshParts()
+	. = ..()
+	gas_temp = 100
+	//starts at 100, should go down to 60
+	for(var/obj/item/stock_parts/micro_laser/laser_part in component_parts)
+		gas_temp -= (laser_part.rating * 5)
+	processing_speed = 6 SECONDS
+	//starts at 6 seconds, should go down to 2 seconds
+	for(var/obj/item/stock_parts/manipulator/manipulator_part in component_parts)
+		processing_speed -= (manipulator_part.rating * (0.5 SECONDS))
+
+/obj/machinery/bluespace_miner/update_overlays()
+	. = ..()
+	cut_overlays()
+	if(panel_open)
+		add_overlay("miner_open")
+	if(machine_stat & (NOPOWER|BROKEN))
+		return
+	add_overlay("miner_on")
+
+/obj/machinery/bluespace_miner/examine(mob/user)
+	. = ..()
+	var/turf/src_turf = get_turf(src)
+	var/datum/gas_mixture/environment = src_turf.return_air()
+	if(environment.temperature >= T20C)
+		. += span_warning("[src] is in a suboptimal environment: TOO HOT TEMPERATURE!")
+	if(environment.return_pressure() <= ONE_ATMOSPHERE)
+		. += span_warning("[src] is in a suboptimal environment: TOO LITTLE PRESSURE!")
+	if(environment.return_pressure() >= (ONE_ATMOSPHERE * 2))
```

Comment:
It shouldn't have both at once, so might as well reduce the checks it does.
```suggestion
	else if(environment.return_pressure() >= (ONE_ATMOSPHERE * 2))
```

Reply:
done